packages/react-core/src/components/NumberInput/__tests__/NumberInput.test.tsx

```diff
@@ -101,72 +101,94 @@ describe('numberInput', () => {
     expect(onChangeMock).not.toHaveBeenCalled();
   });
 
-  test('calls onChange callback when input changes', () => {
+  test('calls onChange callback when input changes', async () => {
     const onChangeMock = jest.fn();
+    const user = userEvent.setup();
+
     render(<NumberInput onChange={onChangeMock}>55</NumberInput>);
 
     const input = screen.getByRole('spinbutton');
-    userEvent.type(input, '55');
+    await user.type(input, '55');
 
     expect(onChangeMock).toHaveBeenCalledTimes(2);
   });
 
-  test('does not call onBlur callback when input does not lose focus', () => {
+  test('does not call onBlur callback when input does not lose focus', async () => {
     const onBlurMock = jest.fn();
+    const user = userEvent.setup();
 
     render(<NumberInput onBlur={onBlurMock}>5</NumberInput>);
 
     const input = screen.getByRole('spinbutton');
-    userEvent.click(input);
+    await user.click(input);
 
     expect(onBlurMock).not.toHaveBeenCalled();
   });
 
-  test('calls onBlur callback when input loses focus', () => {
+  test('calls onBlur callback when input loses focus', async () => {
     const onBlurMock = jest.fn();
+    const user = userEvent.setup();
 
     render(<NumberInput onBlur={onBlurMock}>5</NumberInput>);
 
     const input = screen.getByRole('spinbutton');
-    userEvent.click(input);
-    userEvent.click(document.body);
+    await user.click(input);
+    await user.click(document.body);
 
     expect(onBlurMock).toHaveBeenCalledTimes(1);
   });
 
-  test('removes leading zeros from a positive whole number', () => {
+  test('removes leading zeros from a positive whole number', async () => {
+    const user = userEvent.setup();
+
     render(<NumberInput value={10} onChange={() => {}} />);
-  
+
     const input = screen.getByRole('spinbutton');
-    userEvent.type(input, '{arrowleft}{arrowleft}0');
+
+    await user.click(input);
+
+    // fireEvent is used here do to an issue with the current version userEvent
```

Comment:
*due* to an issue (same as other instances of this comment).

Is there a github issue we can link here pointing to the issue or can the issue be described here in more detail?

Reply:
Also I'll rebase once more and fix the merge conflict